### PR TITLE
docs: Add "configuring APISIX" section to "Installation" docs

### PR DESCRIPTION
**Pull request by navendu-pottekkat**
Signed-off-by: Navendu Pottekkat <navendupottekkat@gmail.com>

### Description

This PR moves the "Configuring APISIX" section to the "Installation" docs from the "Architecture Design" page.

### CODE REVIEW

File: docs/en/latest/installation-guide.md [hunk 1]
```diff
@@ -214,6 +214,48 @@ brew services start etcd
...
+
+APISIX's default configuration can be found in `conf/config-default.yaml` file and it should not be modified. It is bound to the source code and the configuration should only be changed by the methods mentioned above.
+
+:::
+
+:::warning
+
+The `conf/nginx.conf` file is automatically generated and should not be modified.
+
+:::2
```
Comment (hf400159) [inline docs/en/latest/installation-guide.md:257]:
```suggestion
:::
```

Comment (hf400159) [reply docs/en/latest/installation-guide.md:257]: Extra `2`?

Comment (navendu-pottekkat) [reply docs/en/latest/installation-guide.md:257]: Typo!

**Review by juzhiyuan — APPROVED** [review]
It makes sense to me 🙏 cc @tzssangglass to have a confirmation

**Review by tzssangglass — APPROVED** [review]

**Pull request merged**